gateway/crates/federated-server/src/server.rs

```diff
@@ -229,6 +249,11 @@ async fn graceful_shutdown(handle: axum_server::Handle) {
 
     tracing::info!(target: GRAFBASE_TARGET, "Shutting down gracefully...");
     handle.graceful_shutdown(Some(std::time::Duration::from_secs(3)));
+
+    let wg = WaitGroup::new();
```

Comment:
Should be done at the same place `server_runtime.graceful_shutdown()` is called. The `handle.graceful_shutdown` does not wait for requests to finish, in only sends a signal. So you have to add it _after_ the `bind(...).await`